File: C#Part2/HomeworkStringsAndTextProcessing/ParseTags/ParseTags.cs
//You are given a text. Write a program that changes the text in all regions surrounded by the tags <upcase> and </upcase> to upper-case.
//The tags cannot be nested.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class ParseTags
{
    static void Main()
    {
        string input = "We are living in a <upcase>yellow submarine</upcase>. We don't have <upcase>anything</upcase> else.";

        StringBuilder result = new StringBuilder();

        for (int i = 0; i < input.Length; i++)
        {
            if (input[i] == '<')
            {
                i += 8;
                while (input[i] != '<')
                {
                    result.Append(input[i].ToString().ToUpper());
                    i++;
                }
                i += 8;
            }
            else
            {
                result.Append(input[i]);
            }
        }

        Console.WriteLine("The result is: {0}", result);

    }
}

File: 6 JavaScriptBasic/proverka na domashni/StringsJS/04. Parse tags/Parse tags.js
// You are given a text. Write a function that changes the text in all regions:
// <upcase>text</upcase> to uppercase.
// <mixcase>text</mixcase> to mix casing(random).

function toMixCase(str) {
    var replaced = ''
      , i
    for (i = 0; i < str.length; i++)
        replaced += str.charAt(i)[Math.round(Math.random()) ? 'toLowerCase' : 'toUpperCase']()
    return replaced
}
function changeText(str) {
    return str.replace(/<upcase>(.*?)<\/upcase>/g, function (match, p1) {
        return p1.toUpperCase()
    }).replace(/<lowcase>(.*?)<\/lowcase>/g, function (match, p1) {
        return p1.toLowerCase()
    }).replace(/<mixcase>(.*?)<\/mixcase>/g, function (match, p1) {
        return toMixCase(p1)
    })
}

console.log(changeText('We are <mixcase>living</mixcase> in a <upcase>yellow submarine</upcase>. We <mixcase>don\'t</mixcase> have <lowcase>ANYTH<mixcase>living</mixcase>ING</lowcase> else.'))